apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/Query.cs

```diff
@@ -390,11 +390,12 @@ private Cursor CreateCursor(object[] fieldValues, bool before)
                         case string relativePath:
                             // Note: this assumes querying over a single collection at the moment.
                             // Convert to a DocumentReference for the cursor
+                            PathUtilities.ValidateId(relativePath, nameof(fieldValues));
                             value = Collection.Document(relativePath);
                             break;
                         case DocumentReference absoluteRef:
-                            // Just validate
-                            GaxPreconditions.CheckArgument(absoluteRef.IsDescendantOf(Collection), nameof(fieldValues),
+                            // Just validate that the given document is a direct child of the parent collection.
+                            GaxPreconditions.CheckArgument(absoluteRef.Parent.Equals(Collection), nameof(fieldValues),
```

Comment:
It's guaranteed that `Parent` can never be `null`?

Reply:
Yup (and even documented :)

(The parent of a collection can be null, but never the parent of a document.)